src/commands/publish.js

```diff
@@ -1,6 +1,37 @@
 // @flow
 import type {Args, Opts} from '../types';
+import {PError} from '../utils/errors';
+import * as logger from '../utils/logger';
+import * as locks from '../utils/locks';
+import getUnpublishedPackages from '../utils/getUnpublishedPackages';
+import Project from '../Project';
+import * as npm from '../utils/npm';
+import * as yarn from '../utils/yarn';
+
 
 export default async function publish(args: Args, opts: Opts) {
+  const cwd = process.cwd();
+  const project = await Project.init(cwd);
+  const workspaces = await project.getWorkspaces();
+  const packages = workspaces.map(workspace => workspace.pkg);
+
+  try {
+    await locks.lock(packages);
+    const unpublishedPackages = await getUnpublishedPackages(opts);
+    const isUnpublished = workspace => unpublishedPackages.some(pkg => workspace.pkg.config.name === pkg.name);
+    const unpublishedWorkspaces = workspaces.filter(isUnpublished);
```

Comment:
Wasn't super happy with this little dance here
getWorkspaces -> getPackages -> lockPackages -> getUnpublishedPackages -> use those to filter original workspaces.